tests/semantic_analyser.cpp

```diff
@@ -3006,6 +3006,254 @@ TEST_F(semantic_analyser_btf, fentry)
   test("fentry:func_1 { $x = args->a; }", 0);
 }
 
+TEST(semantic_analyser, for_loop_map_one_key)
+{
+  test("BEGIN { @map[0] = 1; for ($kv : @map) { print($kv); } }", R"(
+Program
+ BEGIN
+  =
+   map: @map :: [int64]
+    int: 0 :: [int64]
+   int: 1 :: [int64]
```

Comment:
This indentation looks off.

Reply:
In the AST, the value being assigned to the map is a child of the AssignMapStatement node (shown as "=" by Printer).

This is the tree that we're trying to show:

```
     AssignMapStatement (=)
      /                \
     / LHS              \ RHS
    /                    \
  Map (@map)          Integer (1)
   |
   | key
   |
Integer (0)
```

Reply:
Ah yeah, my bad, I wasn't noticing the equal assignment being the parent.